File: beginner/pagina04/1160/1160.ts
const input = `6
100 150 1.0 0
90000 120000 5.5 3.5
56700 72000 5.2 3.0
123 2000 3.0 2.0
100000 110000 1.5 0.5
62422 484317 3.1 1.0`

//const fs = require('fs')
//const input = fs.readFileSync('/dev/stdin', 'utf-8')
const lines = input.split('\n')

const number_of_test_cases = Number(lines.shift())
const test_cases = lines.map(testcase => testcase.split(' ').map(Number))

type DATA = {
    PA: number, // população inicial A
    PB: number, // população inicial B
    GA: number, // crescimento populacional A
    GB: number, // crescimento populacional B
}

const population_increase = (params: DATA): string => {
    let { PA, PB, GA, GB } = params

    let response = 1;
    for(let anos = 1; PA <= PB && anos < 101; ++anos){
        PA += Math.floor(PA * GA)
        PB += Math.floor(PB * GB)
        response++
    }

    return PA <= PB ? "Mais de 1 seculo." : `${response - 1} anos.`
}

const main = (): void => {
    for(let i = 0; i < number_of_test_cases; i++){
        let PA = test_cases[i][0]
        let PB = test_cases[i][1]
        let GA = test_cases[i][2] / 100
        let GB = test_cases[i][3] / 100

        let result = population_increase({ PA, PB, GA, GB })

        console.log(result)
    }
}

main()

File: beginner/pagina04/1160/1160.py
import math

n = int(input())

def crescimento_popucional(PA, PB, GA, GB):
    response = 1
    for anos in range(1, 101):
        if PA > PB:
            break
        PA += int(PA * GA)
        PB += int(PB * GB)
        response += 1

    return "Mais de 1 seculo" if PA <= PB else f"{response - 1} anos."

for _ in range(n):
    pa, pb, ga, gb = input().split()

    pa = int(pa)
    pb = int(pb)
    ga = float(ga) / 100
    gb = float(gb) / 100

    crescimento = crescimento_popucional(pa, pb, ga, gb)

    print(crescimento)